### PR TITLE
add IDE-extension to improve DX(i18n) in code

### PR DESCRIPTION
**Pull request by NilsJacobsen**
## Description

In order to use the i18n [IDE-extension](https://inlang.com/documentation/apps/ide-extension) of `inlang` a changed to following files:
- Added inlang.config.js 
- Added extensions to recommendations

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Previews

Extract the message to add it to the .json translation files.
image

Inline Annotations to see the language string of the default language:
image

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.

### CODE REVIEW

File: inlang.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	)
+  const { default: standardLintRules } = await env.$import(
+		"https://cdn.jsdelivr.net/npm/@inlang/plugin-standard-lint-rules@3/dist/index.js",
+	)
+
+	return {
+		referenceLanguage: "en-US",
+		plugins: [
+			i18nextPlugin({
+				pathPattern: "./packages/mask/shared-ui/locales/{language}.json",
```
Comment (Jack-Works) [inline inlang.config.js:13]: we have many locales folder, if you configured it like this, it won't work for other monorepo projects

**Comment by NilsJacobsen** [comment]
Hello, @Jack-Works. I have discussed the matter with @inlang and wondered if there is a plan to address it or if a quick fix may be possible.

**Comment by Jack-Works** [comment]
> Hello, @Jack-Works. I have discussed the matter with @inlang and wondered if there is a plan to address it or if a quick fix may be possible.

Our setup also allows the use of multiple i18n files by importing them from different packages.

```ts
import { useSharedI18N } from '@masknet/shared'
import { useLocalI18N } from '../utils/i18n_generated'

export function Comp() {
    
}
```

**Comment by Jack-Works** [comment]
Ah, also we're not using `t('string')`, but `t.string()`.

**Comment by Jack-Works** [comment]
image

We're using a custom setup to improve DX and also type checking.

**Comment by NilsJacobsen** [comment]
Ahh, I then found an old file. The screenshot above is actually out of the `mask` app. 
As far as I understood the [ide-extension](https://inlang.com/documentation/apps/ide-extension#configuration) is configurable so it should be able to create these t.sting() terms.

two things to be figured out:
- parser for ide-extension configuration
- monorepo support for `inlang.config.js`

If we are interested in the ide-extension I can figure that out and keep you updated. We can also make that PR into a draft PR if you want. I think until the end of next week we should get this up and running.

**Comment by Jack-Works** [comment]
> ```ts
> import { useSharedI18N } from '@masknet/shared'
> import { useLocalI18N } from '../utils/i18n_generated'
> 
> export function Comp() {
>     const tShared = useSharedI18N()
>     const t = useLocalI18N()
>     tShared.name() // "XYZ"
>     t.name() // "abc"
> }
> ```

**Comment by Jack-Works** [comment]
> ```ts
> import { useSharedI18N } from '@masknet/shared'
> import { useLocalI18N } from '../utils/i18n_generated'
> 
> export function Comp() {
>     const tShared = useSharedI18N()
>     const t = useLocalI18N()
>     tShared.name() // "XYZ"
>     t.name() // "abc"
> }
> ```

can it handle this?

**Comment by NilsJacobsen** [comment]
Ahh @Jack-Works that may be another challenge. Is there a reason you didn't have it in one language package with namespaces? And then import it from there. Cause if you use them across all apps it feels like there is no reason to sort them in each package. + makes the whole organization of i18n resources a bit easier. 

Let me update the issue I opened at inlang. Honestly, I feel like it should be supported by inlang, but I am unsure if that will be implemented in the next two weeks. Let`s push priority. Feel free to join the discussion:
https://github.com/inlang/inlang/issues/985

**Comment by Jack-Works** [comment]
thanks, @NilsJacobsen, I'll watch that issue.

Yes, our setup is very complex so no IDE plugin can support this. I'm not hoping too much but I'm happy to see if your team can figure out how to support our setup.
It's also ok to give up our complex cases. I'll close this PR for now, but if your produce can provide some support, welcome to open PR again!

**Pull request closed without merging**